### ISSUE TITLE
Improve remediation message for label rule for labels that use aria-labelledby

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#3693]
For the following html (and others that are similar), the `label` rule fails due to a problem in Firefox / NVDA that causes label elements with `aria-labelledby` to not generate an accessible name. In JAWS / Chrome and VO / Safari, the label is correctly given an accessible name.

```html
<div id="foo">hello world</div>

<label for="explicit" aria-labelledby="foo"></label>
<input id="explicit" >  <!-- announces: "Edit has autocomplete; blank" -->
```

However the remediation message for the failure does not help to actually fix the problem as all of the things it says to do already exist on the element. It is wrapped by an explicit label that uses `aria-labelledby` on an element that exists and is not empty. We should try to see if we can clarify the remediation message to say that this is a bug in screen readers and you should not use `aria-labelledby` on label elements for this reason.

> "Fix any of the following:
  Form element does not have an implicit (wrapped) <label>
  Form element does not have an explicit <label>
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute
  Element has no placeholder attribute
  Element's default semantics were not overridden with role="none" or role="presentation""

Related pr: https://github.com/dequelabs/axe-core/pull/3688

**Comment by scottaohara** [comment]
Label elements are meant to be treated as naming prohibited by authors for exactly this reason.  it's not so much a bug in screen readers as much as it's an authoring bug that some browsers / AT account for, and others don't.

**Comment by dylanb** [comment]
@scottaohara are you saying that we should raise an issue that `aria-labelledby` is not allowed to be used on a `<label>` element?

**Comment by scottaohara** [comment]
for the hot minute that ARIA was going to create roles for parity with the label and legend elements, these were specified as being name-prohibited from authors.  While it was eventually decided that there wasn't a need for these roles at this time, the intent to keep these elements as author naming prohibited has remained and is specified as such in the ARIA in HTML spec - and pending future update, there will be a section in the HTML AAM spec that covers this author requirement as well. Support for naming these elements via aria-label/labelledby is - as mentioned in this thread - finicky.  It's not really a bug with screen readers as much as it's just inconsistent exposure of this data by browsers, with those that support largely doing so to handle poor markup choices by developers. There are far better and more robust ways to provide accNames to such elements.

per the example HTML in the OP, using `aria-labelledby` on the label element with no rendered subtree is a rather odd detour to take when using the attribute on the input element itself is the more robust and logical way to name the element.  This actually seems a rather good use case (an input referencing a label but that label attempting to get its name from another element or an aria-label) for the opportunity to provide explicit guidance stating  omething along the lines of: "if you need to name the input and cannot use the label element's subtree, then use aria-label or aria-labelledby on the form control"
